### ISSUE TITLE
useTranslation is the cause of the rerender component

### ISSUE THREAD

**Issue by developer82** [i18next/react-i18next#1636]
## 🐛 Bug Report

When using the `useTranslation` hook the component is re-rendered even if the translation is not used and when there is no change to the language.
I have found a similar reported issue here: https://github.com/i18next/react-i18next/issues/1291 but it was closed with the explanation that there is not issue.
To rule out a situation that it is something in my app that I'm missing, I've created a new react app without anything in it and reproduced the issue.

## To Reproduce

1. Create a new react app using `npx create-react-app my-app`
2. Add the translation as described here: https://react.i18next.com/latest/using-with-hooks
3. Add a `console.log("App component loading");` in App component inside the App function.
4. Once `const { t, i18n } = useTranslation();` is added `"App component loading"` is printed twice in the console. When removed it is ok.

```js
import logo from './logo.svg';
import './App.css';
import { useTranslation } from 'react-i18next';

function App() {
  const { t, i18n } = useTranslation();

  console.log("App component loading");

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
```

## Expected behavior

Component should not re-render by the `useTranslation` hook.

## Your Environment

- *runtime version*: node v18.12.1
- *i18next version*: 22.4.15
- *os*: Windows
- *i18next-browser-languagedetector*: 7.0.1
- *react-i18next*: 12.2.2

**Comment by adrai** [comment]
using an i18next backend like i18next-http-backend will for sure rerender, because the i18next init call is asynchronous and also loading the specific namespace is asynchronous.
If you log the ready flag `const { t, i18n, ready } = useTranslation();` you'll see that the translations are probably not ready.

**Comment by adrai** [comment]
How does your index file look like? Are you using the `<React.StrictMode>` ? https://github.com/i18next/react-i18next/issues/1599#issuecomment-1364741995

**Comment by developer82** [comment]
Yes, using `<React.StrictMode>` (it comes default with a new project and from my understanding it's a good practice to use it - should I not?)

```
import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';

import './i18n';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
```

**Comment by adrai** [comment]
Then it's normal... like said here: it only happens during development: https://github.com/i18next/react-i18next/issues/1599#issuecomment-1365239172

I have no idea how to prevent that.

**Comment by developer82** [comment]
I see. You are right. Tested with a built app and it doesn't happen.
Since this is using in my `App` component and I have some logic there with routing involved this is a bit of a development challenge for me... Thanks.
